lambdas/format_input.py

```diff
@@ -0,0 +1,190 @@
+import datetime
+
+REQUIRED_FIELDS = (
+    "harvest-type",
+    "opensearch-url",
+    "output-bucket",
+    "starting-step",
+    "source",
+    "time",
+)
+
+REQUIRED_HARVEST_FIELDS = ("oai-pmh-host", "oai-metadata-format")
+
+VALID_DATE_FORMATS = ("%Y-%m-%d", "%Y-%m-%dT%H:%M:%SZ")
+
+VALID_HARVEST_TYPES = ("full", "daily")
+
+VALID_STARTING_STEPS = ("harvest", "transform", "load")
+
+
+def lambda_handler(event: dict, context: dict) -> dict:  # noqa
+
+    validate_input(event)
+
+    harvest_end_date = generate_harvest_end_date_string(event["time"])
+    harvest_type = event["harvest-type"]
+    source = event["source"]
```

Comment:
Is this being validated earlier in the process or should it be validated here? I realize there could be sync issues as new sources are added but it seems like we should have a controlled list for this like `harvest-type` and `starting-steps`

Reply:
Yeah I thought about that and I disliked the potential for mismatches across repositories, but now that I'm thinking more about it we could store a list of sources in SSM and use that to validate both here and in transmog. I'll add a separate spike ticket to make a plan for this.

Reply:
I like that approach